### PR TITLE
[RAINCATCH-1388] Filestore tests

### PR DESCRIPTION
**Pull request by paolobueno**
## Motivation
Add unit testing for the server-side file storage module and implementations

## Progress
- [X] Tests for the FileService class
- [x] Tests for the LocalStorage implementation
- [x] Tests for the GridFsStorage implementation
- [x] Update jsdocs

### CODE REVIEW

**Comment by paolobueno** [comment]
ping @JameelB

File: cloud/filestore/src/services/FileService.ts [hunk 1]
```diff
@@ -21,50 +23,44 @@ export const FILE_STORAGE_DIRECTORY = path.join(os.tmpdir(), '/raincatcher-file-
...
-  return deferred.promise;
 }
 
 /**
  * Returns the full path to a file stored with the service
  * @param fileName Name of the file to build a path for, usually the file's id
  */
-export function buildFilePath(fileName) {
+export function buildFilePath(fileName: string, root: string = FILE_STORAGE_DIRECTORY) {
   return path.join(FILE_STORAGE_DIRECTORY, fileName);
```
Comment (JameelB) [inline cloud/filestore/src/services/FileService.ts]: does the `FILE_STORAGE_DIRECTORY` have to be changed to `root`?

**Comment by paolobueno** [comment]
BTW The snyk vulns are fixed since the express version has been updated to match the other packages, and they wouldn't affect a server since the only usage of express from inside the filestore package is to export a router that is run on the server's non-vulnerable express version.

The warning should go away as we publish a new version.

**Review by austincunningham — COMMENTED** [review]
Verified 
```> @raincatcher/filestore@1.1.0-alpha.f71975b7 test /home/austincunningham/repos/raincatcher-core/cloud/filestore
> nyc mocha



  GridFsStorage
    ✓ should write and read back a file from storage (51ms)
    constructor
      ✓ should accept a mongodb connection

  FileService
    createTemporaryStorageFolder
      ✓ should create a default directory
      ✓ should be idempotent
    writeStreamToFile
      ✓ should write data to the file identified by id
    buildFilePath
      ✓ should return a relative path to the root directory
    multerMiddleware
      ✓ should return a middleware
      ✓ should accept other storage implementations


  8 passing (143ms)

-------------------|----------|----------|----------|----------|----------------|
File               |  % Stmts | % Branch |  % Funcs |  % Lines |Uncovered Lines |
-------------------|----------|----------|----------|----------|----------------|
All files          |    85.71 |    58.33 |       75 |    84.62 |                |
 impl              |    80.95 |       50 |    72.73 |    80.49 |                |
  GridFsStorage.ts |    80.95 |       50 |    72.73 |    80.49 |... 56,68,76,77 |
 services          |    92.86 |    66.67 |    77.78 |    91.67 |                |
  FileService.ts   |    92.86 |    66.67 |    77.78 |    91.67 |          42,59 |
-------------------|----------|----------|----------|----------|----------------|```

**Review by austincunningham — APPROVED** [review]

**Pull request merged**